tests/endpoint_test.js

```diff
@@ -0,0 +1,91 @@
+/* eslint-env mocha */
+const mongoose = require('mongoose');
+const chai = require('chai');
+
+require('dotenv').config();
+chai.use(require('chai-http'));
+
+process.env.NODE_ENV = 'test';
+
+const db_util = require('./db_util');
+
+const app = require('../server');
+
+// During the test the env variable is set to test
+
+mongoose.connect(process.env.TEST_CONNECTION_STRING, { useNewUrlParser: true });
+
+const assert = chai.assert;
+
+const request = chai.request(app).keepOpen();
+
+/* eslint-disable prefer-arrow-callback */
+
+// Our parent block
+describe('API', function () {
+    before(function (done) {
+        // Before running suite we populate the db
+        mongoose.connection.once('open', () => {
+            db_util.populateDB()
+                .then(() => done());
+        });
+    });
+
+    describe('/GET Bad API key', function () {
+        it('should reject an incorrect api key', function () {
+            return request.get(`/profile/${process.env.TEST_USER_ID}?token=badkey123`)
+                .then((res) => {
+                    assert.equal(res.status, 401);
+                    assert.isEmpty(res.body);
+                })
+                .catch((err) => {
+                    assert.fail(err);
+                });
+        });
+    });
+
+    describe('/GET Bad User ID', function () {
+        it('should reject a non-existent user', function () {
+            return request.get(`/profile/123456789123456789123456?token=${process.env.TEST_API_KEY}`)
+                .then((res) => {
+                    assert.equal(res.status, 401);
+                    assert.isEmpty(res.body);
+                })
+                .catch((err) => {
+                    assert.fail(err);
+                });
+        });
+    });
+
```

Comment:
Could we add a test for getting a different user's profile with my api-key? We should fail in that case (as per comment above)

Reply:
Both of those cases are handled  in the first 2 tests. And we would get the users contacts via /contacts not /profile which provides all the info we need. And I also have a test for that to ensure we don't leak sensitive data

Reply:
I see, that makes sense.